### DISCUSSION TITLE
How to create a bat file that changes the background?

### DISCUSSION THREAD

**Discussion by Jackgito** [rocksdanister/lively#1633] [Q&A] [answered]
Context: I am trying to create a Python script that changes the wallpaper based on the time of day by executing a bat file.

I made a bat file that was supposed to change the background, but nothing happened. Here are the commands I tried:

livelycu.exe setwp --file "C:\Users\Juise\Desktop\Jutut\Koodi\Dynamic wallpaper\[day.mov](https://day.mov/)"

livelycu.exe setwp --file "[day.mov](https://day.mov/)"

However I did manage to change the wallpaper hue with the bat file, so I don't know why the background doesn't change. Any help is appreciated!

Here's the directory I'm working in:
![image](https://user-images.githubusercontent.com/102518689/225900847-79fdd1f2-d600-41b4-bd1a-4b6a685c35f7.png)

**Comment by rocksdanister** [comment]
Can you create a repository with the python and bat file code?

**Reply by Jackgito** [reply]
Sure, here's the link: https://github.com/Jackgito/DynamicLivelyWallpaper
The video files were too big to upload. And I made some changes so now the video file should be in mp4-format instead of vow.
Thanks for helping me!

**Comment by rocksdanister** [comment, marked as answer]
> https://github.com/Jackgito/DynamicLivelyWallpaper

I was hoping to see code directly to send pull request 😅 

Looks like import is broken now, can only set wallpapers already in gallery.

Here is a working python code that run the commands directly without bat files, place Livelycu.exe in the same folder:

```python
import subprocess

def changeProperty(name, value, monitor):
   runCommand(['setprop', f'--monitor={monitor}', f'--property={name}={value}'])
   return;

def setWallpaper(path, monitor):
   runCommand(['setwp', f'--monitor={monitor}', f'--file={path}'])
   return;
   
def runCommand(args):
    subprocess.run(['Livelycu'] + args)
    print(['Livelycu'] + args)
    return;

setWallpaper('D:\\test\\test clip.mp4', 1)
#changeProperty('hue', 30, 1)
```

Since import is broken `test clip.mp4` should already be present in library.
For the time being add the wallpaper to library first before calling setWallpaper.

**Reply by Jackgito** [reply]
The program works now, though there are a couple of issues. The background change isn't smooth, because instead of instantly changing from day to night background, there is about half a second delay where you can see the default background. Is there any chance to make the transition immediate so there's no "flicker" between the wallpapers? The same thing happens when changing from night to day.

Another problem is that for some reason I have to specify the entire path to the video file, even though it's in the same directory. I would like to share the program with others and I want it to be as user-friendly as possible. 
So instead of this: setWallpaper("C:\\Users\\Juise\\Desktop\\Jutut\\Koodi\\Dynamic wallpaper\\night.mp4", 1)
It should work with this (but it doesn't): setWallpaper("night.mp4", 1)

Here's the program I currently have:
```python
import time
import datetime
import subprocess

def changeProperty(name, value, monitor):
   runCommand(['setprop', f'--monitor={monitor}', f'--property={name}={value}'])
   return

def setWallpaper(path, monitor):
   runCommand(['setwp', f'--monitor={monitor}', f'--file={path}'])
   return
   
def runCommand(args):
    subprocess.run(['Livelycu'] + args)
    print(['Livelycu'] + args)
    return

def main():
    # This is used to prevent unnecessary calls
    dayOrNight = "Unknown"

    while(True):
        current_time = datetime.datetime.now()
        if (current_time.hour >= 20 or current_time.hour < 6) and (dayOrNight == "Unknown" or dayOrNight == "Day"):
            setWallpaper("C:\\Users\\Juise\\Desktop\\Jutut\\Koodi\\Dynamic wallpaper\\night.mp4", 1)
            dayOrNight = "Night"

        elif (current_time.hour < 20 and current_time.hour > 6) and (dayOrNight == "Unknown" or dayOrNight == "Night"):
            setWallpaper("C:\\Users\\Juise\\Desktop\\Jutut\\Koodi\\Dynamic wallpaper\\day.mp4", 1)
            dayOrNight = "Day"

        time.sleep(1)

main()
```

**Reply by rocksdanister** [reply]
>Another problem is that for some reason I have to specify the entire path to the video file, even though it's in the same directory. I would like to share the program with others and I want it to be as user-friendly as possible.
So instead of this: setWallpaper("C:\Users\Juise\Desktop\Jutut\Koodi\Dynamic wallpaper\night.mp4", 1)
It should work with this (but it doesn't): setWallpaper("night.mp4", 1)

File names are not unique, this is by design. 
Files outside Lively appdata folder needs the complete path and for installed wallpapers the complete folderpath where LivelyInfo.json is present.

**Reply by rocksdanister** [reply]
>The background change isn't smooth, because instead of instantly changing from day to night background, there is about half a second delay where you can see the default background. Is there any chance to make the transition immediate so there's no "flicker" between the wallpapers? The same thing happens when changing from night to day.

When wallpaper is changed, the player is closed and new instance is created (this is for stability reasons.) 
You could try turning on "Desktop Picture" in Settings -> Wallpaper; if that does not help not much can be done at the moment.

**Reply by Jackgito** [reply]
Ok I guess this will have to suffice for now. Most of the time the wallpaper is going to change while it's not displayed, so it's not a big deal. Thank you for the help :)
